src/transaction/log_page_buffer.c

```diff
@@ -4099,6 +4099,17 @@ logpb_flush_pages (THREAD_ENTRY * thread_p, LOG_LSA * flush_lsa)
 	  need_wakeup_LFT = true;
 	  nxio_lsa = log_Gl.append.get_nxio_lsa ();
 	}
+
+    if (ats_Gl.is_page_server_connected ())
+        {
+            std::unique_lock<std::mutex> lock (log_Gl.ps_lsa_mutex);
+            log_Gl.ps_lsa_cv.wait (lock, [flush_lsa] { return log_Gl.max_ps_flushed_lsa >= *flush_lsa; });
+
+            if (prm_get_bool_value (PRM_ID_ER_LOG_COMMIT_CONFIRM))
+            {
+                _er_log_debug (ARG_FILE_LINE, "Page server committed LSA = %lld|%d.\n", LSA_AS_ARGS (&log_Gl.max_ps_flushed_lsa));
+            }
+        }
```

Comment:
- Indentantion still looks broken.
- Namespace & lambda syntax are not recognized by indent tool. Disable indent.
- Nitpicking: move the two lines (mutex lock guard and conditional wait) to a log_global function (e.g. `log_globla::wait_flushed_lsa (const log_lsa &flush_lsa)`);